### PR TITLE
Move Crashpad wrapper under starboard

### PR DESCRIPTION
**Pull request by niranjanyardi**
Issue: 365546355

### CODE REVIEW

**Review by dahlstrom-g — DISMISSED** [review]
Can we put this under `chrobalt` instead of modifying
Chromium's `third_party/crashpad/crashpad` tree?

**Review by dahlstrom-g — COMMENTED** [review]
> Can we put this under `chrobalt` instead of modifying
> Chromium's `third_party/crashpad/crashpad` tree?

Since this is a `starboard` dependency
it should probably be under that directory.
As discussed with Niranjan
I've pushed a commit doing that for review.

This is a merge commit restoring the tree
from Cobalt history, so we should not squash it.

File: starboard/crashpad-wrapper/BUILD.gn
Comment (andrewsavage1) [inline starboard/crashpad-wrapper/BUILD.gn:1]: Please rename crashpad-wrapper to crashpad_wrapper or something else that doesn't use a hyphen as python struggles with hyphens in module names (I understand it isn't a python module currently, but it's more future-proof this way).

**Review by andrewsavage1 — CHANGES_REQUESTED** [review]

File: starboard/crashpad-wrapper/wrapper_stub.cc [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  return false;
+}
+
+bool InsertCrashpadAnnotation(const char* key, const char* value) {
+  return false;
+}
+
+}  // namespace wrapper
+}  // namespace crashpad
+}  // namespace third_party
```
Comment (andrewsavage1) [inline starboard/crashpad-wrapper/wrapper_stub.cc:38]: nit: add missing end newline

File: starboard/crashpad-wrapper/BUILD.gn
Comment (dahlstrom-g) [reply starboard/crashpad-wrapper/BUILD.gn:1]: done

File: starboard/crashpad-wrapper/wrapper_stub.cc (hunk 1 above)
Comment (dahlstrom-g) [reply starboard/crashpad-wrapper/wrapper_stub.cc]: This preserves history intact from commit 96e82a55
except the `#include` lines required due to the move
which is the purpose stated in the commit message.
Modifying this line isn't strictly consistent with that
so I mildly prefer not to do it here.

**Comment by niranjanyardi** [comment]
> > Can we put this under `chrobalt` instead of modifying
> > Chromium's `third_party/crashpad/crashpad` tree?
> 
> Since this is a `starboard` dependency it should probably be under that directory. As discussed with Niranjan I've pushed a commit doing that for review.
> 
> This is a merge commit restoring the tree from Cobalt history, so we should not squash it.



> > Can we put this under `chrobalt` instead of modifying
> > Chromium's `third_party/crashpad/crashpad` tree?
> 
> Since this is a `starboard` dependency it should probably be under that directory. As discussed with Niranjan I've pushed a commit doing that for review.
> 
> This is a merge commit restoring the tree from Cobalt history, so we should not squash it.

File: starboard/crashpad-wrapper/BUILD.gn
Comment (andrewsavage1) [reply starboard/crashpad-wrapper/BUILD.gn:1]: The directory name is changed, but all the references still need to be changed

**Review by dahlstrom-g — CHANGES_REQUESTED** [review]
> This is a merge commit restoring the tree from Cobalt history, so we should not squash it.

DO NOT SQUASH

File: starboard/build/config/starboard_target_type.gni [hunk 2]
```diff
@@ -47,9 +47,9 @@ template("starboard_platform_target") {
     ]
     if (sb_is_evergreen_compatible &&
         current_toolchain == starboard_toolchain) {
-      public_deps += [ "//third_party/crashpad/crashpad/wrapper" ]
+      public_deps += [ "//starboard/crashpad-wrapper" ]
```
Comment (niranjanyardi) [inline starboard/build/config/starboard_target_type.gni:50]: should be crashpad_wrapper ?

File: starboard/build/config/starboard_target_type.gni [hunk 3]
```diff
@@ -47,9 +47,9 @@ template("starboard_platform_target") {
     ]
     if (sb_is_evergreen_compatible &&
         current_toolchain == starboard_toolchain) {
-      public_deps += [ "//third_party/crashpad/crashpad/wrapper" ]
+      public_deps += [ "//starboard/crashpad_wrapper" ]
```
[the remark this answers is not in the document]
Comment (niranjanyardi) [reply starboard/build/config/starboard_target_type.gni:50]: I replaced this and added a commit . I'll leave it upto Dana if we want to add it in this commit or later.

With the 2 commits I added , I verified that starboard builds in my branch

[the remark this answers is not in the document]
Comment (andrewsavage1) [reply starboard/build/config/starboard_target_type.gni:50]: @dahlstrom-g I believe this still needs to be addressed with the most recent push (unless you want to address it in a later commit like Niranjan said)

**Review by andrewsavage1 — APPROVED** [review]

**Pull request merged**